src/sql/workbench/services/notebook/browser/sql/sqlSessionManager.ts

```diff
@@ -372,14 +349,17 @@ class SqlKernel extends Disposable implements nb.IKernel {
 	}
 
 	public async disconnect(): Promise<void> {
-		if (this._connectionPath) {
-			if (this._connectionManagementService.isConnected(this._connectionPath)) {
-				try {
-					await this._connectionManagementService.disconnect(this._connectionPath);
-				} catch (err) {
-					this.logService.error(err);
+		if (this._queryRunners.size > 0) {
```

Comment:
If queryRunners length is 0 then forEach will just be a no-op so this check isn't really necessary